Providers/Modules/Plugins/Common/plugin/out_oms_diag.rb

```diff
@@ -0,0 +1,111 @@
+module Fluent
+
+  class OutputOMSDiag < BufferedOutput
+
+    Plugin.register_output('out_oms_diag', self)
```

Comment:
You have a couple options, but pushing the plugins without omsagent.conf changes won't do any harm.

1. Intend for all agent versions to get this diagnostic capability; you will have to test at least a few of our older GA agents, but it will be a handy log collection mechanism to apply to all agents. You will have to append the diagnostic section to omsagent.conf if it isn't there.
2. Only expect agents with the released omsagent.conf changes to use the diagnostic plugin. You don't have to update omsagent.conf through the module, and you don't have to test old agents. It doesn't hurt anything if you push out these plugins to all agents; they just won't be utilized without the changes in omsagent.conf.

It's your design choice. In the past, we haven't really added new sections to the machine's omsagent.conf, so you'll have to be very meticulous in testing that, but it can be done. I would almost recommend making it a separate DSC resource if you change omsagent.conf, though, because nxOMSPlugin is only expected to push plugins/conf to the box, not really re-write anything.